samples/algorithms/repeat-until-success/RepeatUntilSuccess.qs

```diff
@@ -0,0 +1,146 @@
+    /// If this succeeds, the program will have effectively applied an 
+    /// Rz(arctan(2)) gate (also known as V_3 gate) on the target qubit.
+    /// If this fails, the program reruns the circuit up to <limit> times.
+    @EntryPoint()
+    operation CreateQubitsAndApplyRzArcTan2(
+        inputValue : Bool,
+        inputBasis: Pauli,
+        limit: Int
+    ) : (Bool, Result, Int) {
+        using ((auxiliary, resource, target) = (Qubit(), Qubit(), Qubit())) {
+            // Initialize qubits to starting values (|+⟩, |+⟩, |0⟩/|1⟩)
+            InitializeQubits(auxiliary, resource, target, inputBasis, inputValue);
+            let (success, numIter) = ApplyRzArcTan2(auxiliary, resource, target, inputBasis, inputValue, limit);
+            let result = Measure([inputBasis], [target]);
+            // From version 0.12 it is no longer necessary to release qubits in zero state.
+            Reset(target);
+            Reset(resource);
+            Reset(auxiliary);
+            return (success, result, numIter);
+        }
+    }
+
+    /// Apply Rz(arctan(2)) on qubits using repeat until success algorithm
+    operation ApplyRzArcTan2(
+        auxiliary: Qubit,
+        resource: Qubit,
+        target: Qubit,
+        inputBasis: Pauli,
+        inputValue: Bool,
+        limit: Int
+    ) : (Bool, Int) {
+        // Initialize results to One by default.
+        mutable done = false;
+        mutable success = false;
+        mutable numIter = 0;
+
+        repeat {
+            // Assert valid starting states for all qubits
+            AssertMeasurement([PauliX], [auxiliary], Zero, "Auxiliary qubit is not in |+⟩ state.");
+            AssertMeasurement([PauliX], [resource], Zero, "Resource qubit is not in |+⟩ state.");
+            AssertQubitIsInState(target, inputBasis, inputValue);
+
+            // Run Part 1 of the program.
+            let result1 = ApplyAndMeasurePart1(auxiliary, resource);
+            // We'll only run Part 2 if Part 1 returns Zero.
+            // Otherwise, we'll skip and rerun Part 1 again.
+            if (result1 == Zero) { //0X
+                let result2 = ApplyAndMeasurePart2(resource, target);
+                if (result2 == Zero) { //00
+                    set success = true;
+                } else { //01
+                    Z(resource); // Reset resource from |-⟩ to |+⟩
+                    Adjoint Z(target); // Correct effective Z rotation on target
+                }
+            } else { //1X
+                // Set auxiliary and resource qubit back to |+⟩
+                Z(auxiliary);
+                Reset(resource);
+                H(resource);
+            }
+            set done = (success or numIter >= limit);
+            set numIter = numIter + 1;
+        }
+        until (done);
+        return (success, numIter);
+    }
+
+    /// Initialize axiliary and resource qubits in |+⟩, target in |0⟩ or |1⟩
+    operation InitializeQubits(
```

Comment:
In general, Q# style guide suggests putting classical inputs (Pauli and Bool, in this case) before quantum inputs.